secure/storage/src/vault.rs

```diff
@@ -14,36 +14,71 @@ use libra_vault_client::{self as vault, Client};
 /// calls pointers to data keys, Vault has actually a secret that contains multiple key value
 /// pairs.
 pub struct VaultStorage {
-    pub client: Client,
+    client: Client,
+    namespace: Option<String>,
 }
 
 impl VaultStorage {
-    pub fn new(host: String, token: String) -> Self {
+    pub fn new(host: String, token: String, namespace: Option<String>) -> Self {
         Self {
             client: Client::new(host, token),
+            namespace,
         }
     }
 
+    pub fn namespace(&self) -> Option<String> {
+        self.namespace.clone()
+    }
+
     /// Erase all secrets and keys from the vault storage. Use with caution.
     pub fn reset(&self) -> Result<(), Error> {
```

Comment:
sounds a bit scary xD? Should it be behind a test flag?

Reply:
There's a comment in the trait definition to return it back to testing, I'll do it in an ensuing PR, since it needs to be done at the crate feature level and not just at the test cfg level.